src/main/java/sx/blah/discord/util/MessageTokenizer.java

```diff
@@ -0,0 +1,444 @@
+package sx.blah.discord.util;
+
+import sx.blah.discord.api.IDiscordClient;
+import sx.blah.discord.handle.obj.*;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A utility class to traverse through a message's contents and step through tokens like mentions, characters, words,
+ * etc.
+ *
```

Comment:
Do you mind explaining the whole step forward thing here? Beginners may not be totally sure how it works.